Add Zephyr support on WaRP7 Board

diegosueiro opened the pull request:
This patch series is to add support for the WaRP7 board which has an i.MX7 Solo SoC.
The Zephyr is running on the Cortex M4 core and the following features were
validated on this board:

* GPIO: User Switch detection and sensors interrupt
* UART: Zephyr console
* I2C: Communication with fxos8700 and fxas21002 sensors

Other patches included:
* Add all UARTs clock frequency information for i.MX7 devices
* Fix cmake warning about missing id field for fsl,imx7d-i2c
* Check for I2C bus busy before starting a transaction for i.MX devices
* Add pad, clock and gate config for GPIO7 and UART6 for i.MX7 devices

Note that this PR depends on PR #9040 (first commit) and as soon as it got merged this one will be rebased/updated.

diegosueiro committed:
ext/hal/nxp/imx: Add all UARTs clock frequency information

Adds support for retrieving all possible UARTs clock.

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

diegosueiro committed:
dts: Fix cmake warning about missing id field for fsl,imx7d-i2c

Adds the id field in fsl,imx7d-i2c.yaml file

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

diegosueiro committed:
arch/mcimx7_m4: Add i.MX7 Solo Kconfig SoC partnumber define

Adds the proper Kconfig entries for the i.MX7S partnumber presented
on WaRP7 board.

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

diegosueiro committed:
sensors: add WaRP7 board listing for fxos8700 and fxas21002

Update yaml files to include warp7_m4 board in the platform_whitelist
for fxos8700 and fxas21002 sensors.

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

diegosueiro committed:
drivers/i2c/i2c_imx: Check for I2C bus busy before starting transaction

In some cases the minimum time between Stop and Start was not being
considered when starting a new transfer.
This patches adds a checking on the I2C Bus Busy flag before starting
a new transaction.

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

dbkinder commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,341 @@
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
```
delete this blank line and remove the spaces before the next line:

    The default configuration can be found in the defconfig file:
    ``boards/arm/warp7_m4/warp7_m4_defconfig``.

dbkinder commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,341 @@
+
+Download and install `J-Link Tools`_ and `NXP iMX7D Connect CortexM4.JLinkScript`_.
+
+To run Zephyr Binary using J-Link create the following script in order to
```
add a comma after J-Link:    ``using J-Link, create``

Delete ``in order``

dbkinder commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,341 @@
+Download and install `J-Link Tools`_ and `NXP iMX7D Connect CortexM4.JLinkScript`_.
+
+To run Zephyr Binary using J-Link create the following script in order to
+get the Program Counter and Stack Pointer from zephry.bin.
```
zephyr misspelled.  Change to:

    ``zephyr.bin``

dbkinder commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,341 @@
+
+.. code-block:: console
+
+   /usr/bin/JLinkExe -device Cortex-M4 -if JTAG -speed 4000 -autoconnect 1 -jtagconf -1,-1 -jlinkscriptfile iMX7D_Connect_CortexM4.JLinkScript
```
use a continuation line to avoid wrapping:

    /usr/bin/JLinkExe -device Cortex-M4 -if JTAG \
       -speed 4000 -autoconnect 1 -jtagconf -1,-1 \
       -jlinkscriptfile iMX7D_Connect_CortexM4.JLinkScript

dbkinder commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,341 @@
+   /usr/bin/JLinkExe -device Cortex-M4 -if JTAG -speed 4000 -autoconnect 1 -jtagconf -1,-1 -jlinkscriptfile iMX7D_Connect_CortexM4.JLinkScript
+
+The following steps are necessary to run the zephyr.bin:
+1. Put the M4 core in reset
```
add a blank line before starting a list

dbkinder commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,341 @@
+- `Loading Code on Cortex-M4 from Linux for the i.MX 6SoloX and i.MX 7Dual/7Solo Application Processors`_
+- `J-Link iMX7D Instructions`_
+
+.. _WaRP7 Site:
```
As written, only the first two of these references will be displayed in the generated HTML. If you'd like the rest of these to also be shown, add a target-notes directive here (with a blank line after it):

    .. target_note::

diegosueiro replied:
Just the first two should be shown in this Reference session.

dbkinder replied:
OK

diegosueiro commented:
@dbkinder ,

Thanks for the review.
All comments should be addressed.

diegosueiro commented on boards/arm/warp7_m4/doc/warp7_m4.rst:
```diff
@@ -0,0 +1,346 @@
++------------+-----------------------+------------------------+-----------------------+----------------------+
+
+
+References
```
I think this and the one at line 306 are not a heading since they are references only about the memory mapping and debugging.
Do you think I need to put as just a normal paragraph?

diegosueiro replied:
@dbkinder,

Any suggestion?

diegosueiro committed:
arch/mcimx7_m4: Add pad, clock and gate config for GPIO7 and UART6

Adds the necessery configuration for using the GPIO7 and UART6 on
i.MX7 platforms.

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

diegosueiro committed:
boards: Add support for WaRP7 board

This patch is to add support for the WaRP7 board which has an i.MX7
Solo SoC.
The Zephyr is running on the Cortex M4 core and the following features
were validated on this board:

* GPIO: User Switch detection and sensors interrupt
* UART: Zephyr console
* I2C: Communication with fxos8700 and fxas21002 sensors

Signed-off-by: Diego Sueiro <diego.sueiro@gmail.com>

dbkinder reviewed (approved):
I'm OK with two references subtitles.
